File: implementations/dotnet/EmailInquire/Validators/KnownInvalidDomain.cs
using System.Linq;

namespace EmailInquire.Validators
{
    /// <summary>
    ///     Validation over known invalid domains list
    /// </summary>
    public class KnownInvalidDomain : ValidatorBase<KnownInvalidDomain>
    {
        public override Response Validate()
        {
            return KnownInvalidDomains?.Value.Contains(Domain) ?? false ? Response.Invalid(Email) : Response.Undefined;
        }
    }
}

File: implementations/ruby/lib/email_inquire/validator/known_invalid_domain.rb
# frozen_string_literal: true

require "email_inquire/validator/base"

module EmailInquire
  module Validator
    class KnownInvalidDomain < Base

      DOMAINS = load_data("known_invalid_domains").freeze

      def validate
        response.invalid! if DOMAINS.include?(domain)
      end

    end
  end
end
